Examples/Swift/ManageOfflineRegionsExample.swift

```diff
@@ -0,0 +1,220 @@
+        var pack = MGLOfflineStorage.shared.packs
+        var progressPercentage: Float?
+
+        override func viewDidLoad() {
+
+            super.viewDidLoad()
+
+            mapView = MGLMapView(frame: CGRect(x: 0, y: 0, width: view.bounds.width, height: view.bounds.height / 1.5), styleURL: MGLStyle.darkStyleURL)
+            mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+            mapView.tintColor = .gray
+            mapView.delegate = self
+
+            tableView.frame = CGRect(x: 0, y: mapView.frame.maxY, width: view.bounds.width, height: view.bounds.height / 3.5)
+            tableView.delegate = self
+            tableView.dataSource = self
+            tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
+            tableView.rowHeight = 50
+            
+            downloadButton.backgroundColor = UIColor.white
+            downloadButton.setTitleColor(.black, for: .normal)
+            downloadButton.setTitle("Download Region", for: .normal)
+            downloadButton.addTarget(self, action: #selector(startOfflinePackDownload), for: .touchUpInside)
+            downloadButton.layer.cornerRadius = view.bounds.width / 30
+            downloadButton.translatesAutoresizingMaskIntoConstraints = false
+
+            view.addSubview(mapView)
+            view.addSubview(tableView)
+            view.addSubview(downloadButton)
+            mapView.setCenter(CLLocationCoordinate2D(latitude: 22.27933, longitude: 114.16281),
+
+                              zoomLevel: 13, animated: false)
+
+            // Setup offline pack notification handlers.
+            NotificationCenter.default.addObserver(self, selector: #selector(offlinePackProgressDidChange), name: NSNotification.Name.MGLOfflinePackProgressChanged, object: nil)
+            NotificationCenter.default.addObserver(self, selector: #selector(offlinePackDidReceiveError), name: NSNotification.Name.MGLOfflinePackError, object: nil)
+            NotificationCenter.default.addObserver(self, selector: #selector(offlinePackDidReceiveMaximumAllowedMapboxTiles), name: NSNotification.Name.MGLOfflinePackMaximumMapboxTilesReached, object: nil)
+            installConstraints()
+        }
+
+        func installConstraints() {
+
+            if #available(iOS 11.0, *) {
+                NSLayoutConstraint.activate([
+                    downloadButton.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+                    downloadButton.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -170),
+                    downloadButton.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor)
+                ])
+            } else {
+                // Fallback on earlier versions
+            }
+
+        }
+
+        override func viewDidDisappear(_ animated: Bool) {
+
+            super.viewDidDisappear(animated)
+            // When leaving this view controller, suspend offline downloads.
```

Comment:
I think we should instead delete all the offline packs, since they will continue to exists in the database forever until the app is deleted/reinstalled.